src/plugins/intel_gna/src/transformations/remove_in_out_processing.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "gna_data_types.hpp"
+#include "openvino/pass/graph_rewrite.hpp"
+
+namespace ov {
+namespace intel_gna {
+namespace pass {
+
+/**
+ * @brief Remove Transpose/Gather layers connected to Inputs and create pre-processing model
+ * to support input pre-processing on CPU.
+ * Inserts Reshape layer instead of Transpose/Gater layer to avoid changing of the shapes.
+ * @param output_subgraphs Map where pre-processing model for each input will be saved
+ *
+ * Searches for the following pattern
+ *     Any input layer
+ *           |
+ *    Transpose/Gather
+ *           |
+ *        Any layer
+ *
+ *   and transforms it to
+ *     Any input layer
+ *           |
+ *        Reshape
+ *           |
+ *        Any layer
+ */
+class RemoveInputsProcessing : public ov::pass::ModelPass {
+public:
+    OPENVINO_RTTI("RemoveInputsProcessing", "0");
+    RemoveInputsProcessing(ov::intel_gna::PrePostProcessModels* input_subgraphs = nullptr)
+        : m_input_subgraphs(input_subgraphs) {}
+    bool run_on_model(const std::shared_ptr<ov::Model>& model) override;
+
+private:
+    ov::intel_gna::PrePostProcessModels* m_input_subgraphs;
```

Comment:
wouldn't be better to store weak_ptr?

Reply:
good comment, but here it is not important, I prefer plain pointer.